spec/controllers/invitations_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'spec_helper'
+
+describe InvitationsController, type: :controller do
+  describe "GET new" do
+    it "sets @invitation to a new invitation" do
+      set_current_user
+      get :new
+      expect(assigns(:invitation)).to be_instance_of(Invitation)
+    end
+    
+    it_behaves_like "requires sign in" do
+      let(:action) {get :new}
+    end
+   
+  end
+
+  describe "POST create" do
+    it_behaves_like "requires sign in" do
+     let(:action) { post :create}
+    end
+
+    context "with valid input" do
+
+     after {ActionMailer::Base.deliveries.clear}
+
+     it "redirects to the invitation new page" do
+      set_current_user
+        post :create, invitation: { recipient_name: "Joe Smith", recipient_email: "joe@example.com", message: "Hey join Myflix!"}
+        expect(response).to redirect_to new_invitation_path
+     end
+     it "creates an invitation" do
+       set_current_user
+        post :create, invitation: { recipient_name: "Joe Smith", recipient_email: "joe@example.com", message: "Hey join Myflix!"}
+        expect(Invitation.count).to eq(1)
+     end
+      it "sends an email to the recipient" do
+        set_current_user
+        post :create, invitation: { recipient_name: "Joe Smith", recipient_email: "joe@example.com", message: "Hey join Myflix!"}
+        expect(ActionMailer::Base.deliveries.last.to).to eq(['joe@example.com'])
+      end
+      
+      it "sets the flash success message" do
+        set_current_user
```

Comment:
You could eventually refactor `set_current_user` from all these `it` specs and put it to a `before` block.